### PR TITLE
Added callout about failed builds

### PR DESCRIPTION
**Pull request by anenadic**
Addresses #124  and #136.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -69,11 +69,11 @@ This indicates that the maintainers will welcome pull requests fixing such issue
 
 Current maintainers of this lesson (in alphabetical order) are:
 
-* [Matthew Bluteau][matthew-bluteau]
+* [Matthew Bluteau][matthew-bluteau] - Lead Maintainer for the period 1 May 2024 - 30 April 2024
```
Comment (steve-crouch) [inline README.md:72]: Probably should be until end-October 2024?

```suggestion
* [Matthew Bluteau][matthew-bluteau] - Lead Maintainer for the period 1 May 2024 - 31 October 2024
```

File: _episodes/23-continuous-integration-automated-testing.md [hunk 2]
```diff
@@ -410,4 +410,26 @@ which potentially saves us a lot of time waiting for testing results.
 Overall, this approach allows us to massively scale our automated testing
 across platforms we wish to test.
 
+> ## Failed CI Builds
+> A CI build can fail when, e.g. a used Python package no longer supports a particular version of 
+> Python indicated in a GitHub Actions CI build matrix. In this case, the solution is either to 
+> downgrade the package version (and not use the latest one like we have been doing in this course),
+> or upgrade the Python version in the build matrix.
```
Comment (steve-crouch) [inline _episodes/23-continuous-integration-automated-testing.md:417]: Could add a comment on what the best solution is in this situation, i.e. upgrade where possible, as opposed to downgrade.

**Review by steve-crouch — CHANGES_REQUESTED** [review]
Just a minor date change for Matt's tenure as lead, and a few other suggestions.

File: _episodes/23-continuous-integration-automated-testing.md (hunk 2 above)
Comment (anenadic) [reply _episodes/23-continuous-integration-automated-testing.md]: Done!

**Review by steve-crouch — APPROVED** [review]

**Pull request merged**